lib/cfd_trading/cfd_order_detail.dart

```diff
@@ -176,32 +176,7 @@ class _CfdOrderDetailState extends State<CfdOrderDetail> {
                             ),
                             Visibility(
                               visible: confirm,
-                              child: ElevatedButton(
-                                  onPressed: () async {
-                                    try {
-                                      await api.settleCfd(cfd: cfd, offer: offer);
-                                      FLog.info(text: "Successfully settled cfd.");
-
-                                      // switch index to cfd overview tab
-                                      cfdTradingChangeNotifier.selectedIndex = 1;
-                                      // refreshing cfd list after cfd has been closed
-                                      // will also implicitely propagate the index change
-                                      await cfdTradingService.refreshCfdList();
-
-                                      // navigate back to the trading route where the index has already been propagated
-                                      context.go(CfdTrading.route);
-                                    } on FfiException catch (error) {
-                                      FLog.error(
-                                          text: 'Failed to settle CFD: ' + error.message,
-                                          exception: error);
-
-                                      ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
-                                        backgroundColor: Colors.red,
-                                        content: Text("Failed to settle cfd"),
-                                      ));
-                                    }
-                                  },
-                                  child: const Text('Confirm')),
+                              child: AsyncButton(onPressedFunction: settleCfd, label: 'Confirm'),
```

Comment:
nit: we don't have to put the offer as late state variable.

```
AsyncButton(onPressedFunction: () { settleCfd(offer) }, label: 'Confirm')
```

Reply:
Fair enough, I'll change that :)

Reply:
Note: I also adapted this to passing parameters rather than adding fields to the state class for `cfd_order_confirmation`